include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -3998,6 +3998,129 @@ __pattern_swap(_ExecutionPolicy&& __exec, _ForwardIterator1 __first1, _ForwardIt
                            __is_vector);
 }
 
+//------------------------------------------------------------------------
+// shift_left
+//------------------------------------------------------------------------
+
+template <class _ForwardIterator>
+_ForwardIterator
+__brick_shift_left(_ForwardIterator __first, _ForwardIterator __last,
+                   typename std::iterator_traits<_ForwardIterator>::difference_type __n,
+                   /*is_vector=*/::std::false_type) noexcept
+{
+#if __cplusplus >= 202000L
+    return ::std::shift_left(__first, __last, __n);
+#else
+    //If (n > 0 && n < m), returns first + (m - n). Otherwise, if n  > 0, returns first. Otherwise, returns last.
+    if (__n <= 0)
+        return __last;
+
+    //seek for (first + n)
+    auto __it = __first;
+    for (; --__n >= 0; ++__it)
+        if (__it == __last) // n >= last - first;
+            return __first;
+
+    //Moving the rest elements from a position number n to the begin of the sequence.
+    for (; __it != __last; ++__it, ++__first)
+        *__first = ::std::move(*__it);
+
+    return __first;
+#endif
+}
+
+template <class _ForwardIterator>
+_ForwardIterator
+__brick_shift_left(_ForwardIterator __first, _ForwardIterator __last,
+                   typename ::std::iterator_traits<_ForwardIterator>::difference_type __n,
+                   /*is_vector=*/::std::true_type) noexcept
+{
+    //If (n > 0 && n < m), returns first + (m - n). Otherwise, if n  > 0, returns first. Otherwise, returns last.
+    if (__n <= 0)
```

Comment:
Consider to create the common part for `__brick_shift_left` and `__pattern_shift_left`